packages/explat-client/src/internal/test/timing.ts

```diff
@@ -25,21 +27,33 @@ const delayedValue = < T >( value, delayMilliseconds ): Promise< T > =>
 
 describe( 'timeoutPromise', () => {
 	it( 'should resolve promises below the timeout', async () => {
-		await expect( Timing.timeoutPromise( new Promise( ( res ) => res( 123 ) ), 1 ) ).resolves.toBe(
-			123
-		);
-		await expect( Timing.timeoutPromise( delayedValue( 123, 1 ), 2 ) ).resolves.toBe( 123 );
+		const promise1 = Timing.timeoutPromise( new Promise( ( res ) => res( 123 ) ), 1 );
+		jest.advanceTimersToNextTimer();
+		jest.advanceTimersToNextTimer();
```

Comment:
This particular test should need only one `advance` call, is that right? There is only one `setTimeout` call inside the `timeoutPromise` helper. Only when there's also a `delayedValue` call do you need a second call.

This is also one thing I don't like about this approach: the number of `advance` calls needed depends on the internal implemenation. `advanceTimersByTime` would be more universal.

Reply:
Yeah that makes sense :)